numpy/ma/core.py

```diff
@@ -7400,22 +7401,195 @@ def mask_rowcols(a, axis=None):
       fill_value=1)
 
     """
-    a = array(a, subok=False)
+    a = asarray(a)
+    # rather than generalizing this weird API to ND, have the user call the
+    # simpler API
     if a.ndim != 2:
-        raise NotImplementedError("mask_rowcols works for 2D arrays only.")
+        raise NotImplementedError(
+            "mask_rowcols works for 2D arrays only, use mask_extend_axis "
+            "instead (which takes a different axis argument)."
+        )
+    # compatibility logic to accept the weird input that was allowed before
+    axes = ()
+    if not axis:
+        axes += (1,)
+    if axis in [None, 1, -1]:
+        axes += (0,)
+    return mask_extend_axis(a, axes)
+
+
+def mask_extend_axis(a, axis):
+    """
+    Extend masked values along 1d slices.
+
+    If any item is masked, then that entire 1d slice is masked.
+
+    .. versionadded:: 1.19.0
+
+    Parameters
+    ----------
+    a : array_like, MaskedArray
+        The array to mask.  If not a MaskedArray instance (or if no array
+        elements are masked).  The result is a MaskedArray with `mask` set
+        to `nomask` (False).
+    axis : int or tuple of ints, optional
+        Axis along which to extend the mask. If ``None``, do so along
+        all axes.
+
+    Returns
+    -------
+    a : MaskedArray
+        A modified version of the input array, masked depending on the value
+        of the `axis` parameter.
+
+    See Also
+    --------
+    mask_rows : Alias for ``mask_extend_axis(a, axis=-2)``
+    mask_cols : Alias for ``mask_extend_axis(a, axis=-1)``
+
+    Examples
+    --------
+    >>> import numpy.ma as ma
+    >>> a = np.zeros((3, 3, 3), dtype=int)
+    >>> a[1, 1, 1] = 1
+    >>> a
+    array([[[0, 0, 0],
+            [0, 0, 0],
+            [0, 0, 0]],
+
+           [[0, 0, 0],
+            [0, 1, 0],
+            [0, 0, 0]],
+
+           [[0, 0, 0],
+            [0, 0, 0],
+            [0, 0, 0]]])
```

Comment:
@mattip: What did we decide about blank lines in docstring examples?

Reply:
Bank lines are markers for the end of a docstring, so the "expected" is truncated. Why hasn't this come up before, has something changed in the formatting here?

Reply:
I'm sure this has come up before, I just forgot the resolution.

Is the solution to omit the blank lines, or have we been using `<BLANKL N NE>`?

Reply:
We need to add `    <BLANKLINE>` with the appropriate indentation (4 spaces). This is vanilla [doctests](https://docs.python.org/3.7/library/doctest.html#doctest.DONT_ACCEPT_BLANKLINE), we don't massage these tests in any way.

Reply:
In Scipy, we gave up strict doctest compatibility for better readability. `refguide-check.py` allows also omitting the blank lines, and I guess Numpy is nowadays also running CI using it?

Reply:
@pv's comment was exactly what I was trying to get at

Reply:
I tried eliminating the blank lines from the docstring, refguide-check was not happy. Maybe the two implementations have diverged?

Reply:
Hmm, iirc `doctest.NORMALIZE_WHITESPACE` is what does it, but I'm not sure right now.

Reply:
Maybe someday we will move to astropy's [pytest-doctestplus](https://github.com/astropy/pytest-doctestplus) instead of refguide-check